xcdat/axis.py

```diff
@@ -0,0 +1,243 @@
+from typing import List, Literal, TypedDict
+
+import numpy as np
+import xarray as xr
+
+from xcdat.xcdat import logger
+
+
+@xr.register_dataset_accessor("axis")
+class AxisAccessor:
+    """A class used to represent an AxisAccessor (xarray dataset accessor)
+
+    :param xarray_obj: The Dataset object to be extended
+    :type xarray_obj: xr.Dataset
+
+    Examples
+    --------
+    Get bounds:
+
+    >>> from xcdat import axis
+    >>> ds = xr.open_dataset("file_path")
+    >>> # Get generated bounds
+    >>> lat_bnds = ds.axis.get_bounds('lat', generate=True)
+    >>> lon_bnds = ds.axis.get_bounds('lon', generate=True)
+    >>> # Get existing bounds
+    >>> lat_bnds = ds.axis.get_bounds('lat')
+    >>> lon_bnds = ds.axis.get_bounds('lon')
+    """
+
+    # Type annotation definitions
+    Axis = Literal["lat", "lon"]
+    AxesMapValue = TypedDict(
+        "AxesMapValue", {"coords": List[str], "bounds_vars": List[str]}
+    )
+    AxesMap = TypedDict("AxesMap", {"lat": AxesMapValue, "lon": AxesMapValue})
+
+    # Mapping of generic axes names to coordinates and bounds variables in the Dataset
+    axes_map: AxesMap = {
+        "lat": {
+            "coords": ["lat", "latitude"],
+            "bounds_vars": ["lat_bnds", "latitude_bnds"],
+        },
+        "lon": {
+            "coords": ["lon", "longitude"],
+            "bounds_vars": ["lon_bnds", "longitude_bnds"],
+        },
+    }
+
+    def __init__(self, xarray_obj: xr.Dataset):
+        self._dataset = xarray_obj
+
+    def get_bounds(self, axis: Axis, generate: bool = False) -> xr.DataArray:
```

Comment:
Will be defaulting generate to `True` to always generate bounds if they do not exist, and updating param name for clarity